pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -443,6 +451,33 @@ func (c *ruleCache) processPodUpdate(e interface{}) {
 	}
 }
 
+// processExternalEntityUpdate will be called when ExternalNodeController publishes an ExternalEntity update event.
+// It finds out AppliedToGroups that contain this ExternalNode converted ExternalEntity and triggers reconciliation
+// of related rules.
+// It can enforce NetworkPolicies to ExternalEntities after ExternalEntityInterface is realised in the interface store.
+func (c *ruleCache) processExternalEntityUpdate(e interface{}) {
+	externalEntityRef := e.(v1beta.ExternalEntityReference)
+	c.appliedToSetLock.RLock()
+	defer c.appliedToSetLock.RUnlock()
+	externalEntityEquals := func(expEntity *v1beta.ExternalEntityReference, actEntity v1beta.ExternalEntityReference) bool {
+		if expEntity.Name != actEntity.Name {
+			return false
+		}
+		if expEntity.Namespace != actEntity.Namespace {
+			return false
+		}
+		return true
+	}
+	for group, memberSet := range c.appliedToSetByGroup {
+		for _, m := range memberSet.Items() {
+			if externalEntityEquals(m.ExternalEntity, externalEntityRef) {
```

Comment:
I feel not necessary to define a func for this, but up to you.

Reply:
Thanks for pointing out and I agree that this is not necessary. I have changed to use reflect.DeepEqual(m.ExternalEntity, externalEntityRef) to check this part.